lambdas/data-migration2/src/pdrs.ts

```diff
@@ -0,0 +1,133 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import Logger from '@cumulus/logger';
+import {
+  getRecordCumulusId,
+  CollectionPgModel,
+  PdrPgModel,
+  PostgresExecutionRecord,
+  PostgresPdr,
+  ProviderPgModel,
+  tableNames,
+} from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import { RecordAlreadyMigrated } from '@cumulus/errors';
+
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/pdrs' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+/**
+ * Migrate PDR record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migratePdrRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(dynamoRecord, schemas.pdr);
+
+  const collectionPgModel = new CollectionPgModel();
+  const pdrPgModel = new PdrPgModel();
+  const providerPgModel = new ProviderPgModel();
+
+  const existingRecord = await pdrPgModel.get(knex, { name: dynamoRecord.pdrName });
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`PDR name ${dynamoRecord.pdrName} was already migrated, skipping.`);
+  }
+  const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+    knex,
+    { name: dynamoRecord.collectionId }
```

Comment:
I don't think this will ever return anything? The `collectionId` from dynamo is a concatenation of a collection name and version something like `name___version`. So you need to deconstruct that into a name and version, then use both of those keys to look up a collection.

Reply:
Ah, that's true.